classes/class-wc-connect-error-notice.php

```diff
@@ -53,10 +53,23 @@ private function show_notice() {
 			$link_status = admin_url( 'admin.php?page=wc-status&tab=connect' );
 			$link_dismiss = add_query_arg( array( 'wc-connect-error-notice' => 'disable' ) );
 
-			$message = sprintf(
-				__( 'An error occurred in WooCommerce Services. Details are logged <a href="%s">here</a>.', 'woocommerce-services' ),
-				$link_status, $link_dismiss
-			);
+			$error = $this->notice_enabled();
+			if ( is_wp_error( $error ) && 'product_missing_weight' === $error->get_error_code() ) {
+				$matches = array();
+				preg_match( '/ID:\s([0-9]+)/', $error->get_error_message(), $matches );
+				$product_id = $matches[1];
+				$product_name = WC_Connect_Compatibility::instance()->get_product_name( wc_get_product( $product_id ) );
+
+				$message = sprintf(
+					__( 'An error occurred in WooCommerce Services: <strong><a href="%s">%s</a> did not include a weight.</strong> Shipping rates cannot be calculated.', 'woocommerce-services' ),
```

Comment:
Let's use `wp_kses()` to restrict what HTML we allow in the translated string, like: https://github.com/Automattic/woocommerce-services/blob/master/classes/class-wc-connect-paypal-ec.php#L135

Reply:
Done in https://github.com/Automattic/woocommerce-services/pull/1380/commits/3acb4d389ba63d88e21e6b7c830051bc603e4b05